src/compiler/performance.ts

```diff
@@ -101,22 +95,25 @@ namespace ts.performance {
      * @param cb The action to perform for each measure
      */
     export function forEachMeasure(cb: (measureName: string, duration: number) => void) {
-        measures.forEach((measure, key) => {
-            cb(key, measure);
+        perfEntryList?.getEntriesByType("measure").forEach(entry => {
+            cb(entry.name, entry.duration);
         });
     }
 
     /** Enables (and resets) performance measurements for the compiler. */
     export function enable() {
-        counts = new Map<string, number>();
-        marks = new Map<string, number>();
-        measures = new Map<string, number>();
-        enabled = true;
-        profilerStart = timestamp();
+        if (!enabled) {
+            perfHooks ||= tryGetNativePerformanceHooks() || ShimPerformance?.createPerformanceHooksShim(timestamp);

```

Comment:
We can disable/reenable `ts.performance` on an as-need basis, but we don't want to recreate the `PerformanceObserver` and `list` every time, since that gets handled for you when you call `.disconnect()` on the observer.

Reply:
I decided to merge them by replacing `enabled` with a `performanceImpl` value that stores the current `Performance` object.